x/tokenfactory/types/msgs.go

```diff
@@ -116,7 +116,7 @@ func NewMsgBurnFrom(sender string, amount sdk.Coin, _ string) *MsgBurn {
 
 func (m MsgBurn) Route() string { return RouterKey }
 func (m MsgBurn) Type() string  { return TypeMsgBurn }
-func (m MsgBurn) ValidateBasic() error {
+func (m MsgBurn) Validate() error {
```

Comment:
no, we cannot just to that because [if the address is empty the sender address is used](https://github.com/neutron-org/neutron/blob/1c2388b321f215a024923aefc49cb7a929a8dfad/x/tokenfactory/keeper/msg_server.go#L108-L110). we should do the check only if the sender addr is not an empty string

Reply:
Done